### PR TITLE
[du] updating based on API changes from 1dot7

### PR DESCRIPTION
**Pull request by tacastillo**
## Summary & Motivation

based on the API changes listed here https://www.notion.so/dagster/Proposal-of-DU-updates-for-1-7-2bd3ec68fc8b41dfa90099ade8d71a96

### CODE REVIEW

**Comment by tacastillo** [comment]
* **#21725** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21725?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tacastillo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/dagster-university/pages/dagster-essentials/lesson-8/adding-partitions-to-assets.md [hunk 1]
```diff
@@ -43,35 +43,38 @@ To add the partition to the asset:
    )
    ```
 
-3. In Dagster, the `context` argument provides you with metadata about the current materialization. To access it, include it as the first argument in the asset definition function:
+3. In Dagster, the `context` argument provides you with metadata about the current materialization. To access it, include it as the first argument in the asset definition function. You can enable typehinting for this by importing `AssetExecutionContext` from `dagster` and adding it to the function signature. For example, the updated asset definition should look like this:
```
Comment (cmpadden) [inline docs/dagster-university/pages/dagster-essentials/lesson-8/adding-partitions-to-assets.md:46]:
```suggestion
3. In Dagster, the `context` argument provides you with metadata about the current materialization. To access it, include it as the first argument in the asset definition function. You can enable type hinting for this by importing `AssetExecutionContext` from `dagster` and adding it to the function signature. For example, the updated asset definition should look like this:
```

**Comment by cmpadden** [comment]
Corresponding project pull request for linkage - https://github.com/dagster-io/project-dagster-university/pull/25/files

**Review by erinkcochran87 — APPROVED** [review]
I'm good with this, assuming Colton is too!

**Comment by cmpadden** [comment]
Yup, @tacastillo just one tiny comment on if _ typehinting_ is one or two words, other than that lgtm.

**Review by cmpadden — APPROVED** [review]

**Comment by tacastillo** [comment]
### Merge activity

* **May 13, 1:34 PM EDT**: @tacastillo started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/21725).

**Pull request merged**